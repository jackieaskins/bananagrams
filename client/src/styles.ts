import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    closeButton: {
      color: 'white',
      position: 'fixed',
      right: 0,
      top: 0,
      zIndex: theme.zIndex.drawer + 2,
    },
    drawer: {
      [theme.breakpoints.up('sm')]: {
        width: theme.spacing(8) + 1,
      },
      width: theme.spacing(7) + 1,
    },
    drawerPaper: {
      [theme.breakpoints.up('sm')]: {
        width: theme.spacing(8) + 1,
      },
      width: theme.spacing(7) + 1,
    },
    invalidDrop: {
      backgroundColor: theme.palette.error.main,
      color: theme.palette.error.contrastText,
      opacity: 0.5,
    },
    transparentPaper: {
      backgroundColor: 'transparent',
    },
    validDrop: {
      backgroundColor: theme.palette.success.main,
      color: theme.palette.success.contrastText,
      opacity: 0.5,
    },
  })
);
