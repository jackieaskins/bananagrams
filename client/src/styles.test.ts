import { useStyles } from './styles';

jest.mock('@material-ui/core/styles', () => ({
  createStyles: (values: any): any => values,
  makeStyles: (fn: any): any =>
    fn({
      zIndex: {
        drawer: 0,
      },
      spacing: (n: number): number => n,
      breakpoints: {
        up: (size: string): string => `breakpoints-up-${size}`,
      },
      palette: {
        error: {
          main: '#error-main',
          contrastText: '#error-contrast-text',
        },
        success: {
          main: '#success-main',
          contrastText: '#success-contrast-text',
        },
      },
    }),
}));

describe('styles', () => {
  describe('useStyles', () => {
    test('contains expected values', () => {
      expect(useStyles).toMatchSnapshot();
    });
  });
});
