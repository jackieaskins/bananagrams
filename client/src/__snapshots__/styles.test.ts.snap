// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles useStyles contains expected values 1`] = `
Object {
  "closeButton": Object {
    "color": "white",
    "position": "fixed",
    "right": 0,
    "top": 0,
    "zIndex": 2,
  },
  "drawer": Object {
    "breakpoints-up-sm": Object {
      "width": 9,
    },
    "width": 8,
  },
  "drawerPaper": Object {
    "breakpoints-up-sm": Object {
      "width": 9,
    },
    "width": 8,
  },
  "invalidDrop": Object {
    "backgroundColor": "#error-main",
    "color": "#error-contrast-text",
    "opacity": 0.5,
  },
  "transparentPaper": Object {
    "backgroundColor": "transparent",
  },
  "validDrop": Object {
    "backgroundColor": "#success-main",
    "color": "#success-contrast-text",
    "opacity": 0.5,
  },
}
`;
